backend/EasyMeets.Core/EasyMeets.Core.BLL/Services/AvailabilityService.cs

```diff
@@ -1,23 +1,119 @@
+using EasyMeets.Core.Common.DTO.Availability;
 using EasyMeets.Core.DAL.Context;
+using Microsoft.EntityFrameworkCore;
 using EasyMeets.Core.Common.DTO.Availability.NewAvailability;
 using EasyMeets.Core.DAL.Entities;
+using EasyMeets.Core.Common.Enums;
+using EasyMeets.Core.Common.DTO.User;
+using EasyMeets.Core.Common.DTO.Location;
 
 namespace EasyMeets.Core.BLL.Services
 {
     public class AvailabilityService : BaseService, IAvailabilityService
     {
-        public AvailabilityService(EasyMeetsCoreContext context, IMapper mapper) : base(context, mapper)
+        public AvailabilityService(EasyMeetsCoreContext context, IMapper mapper) : base(context, mapper) { }
+        public async Task<ICollection<AvailabilitySlotsGroupByTeamsDto>> GetAllAvailabilitySlotsGroupByTeamsAsync(int id)
         {
+            var teamsWithSlots = await _context.Teams
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Members)
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Location)
+                .Include(x => x.AvailabilitySlots)
+                    .ThenInclude(x => x.Author)
+                .Where(x => x.AvailabilitySlots.Any(x => x.Type == SlotType.Team) && x.AvailabilitySlots.Any(x => x.Members.Any(x => x.UserId == id)))
+                .Select(x =>
+                    new AvailabilitySlotsGroupByTeamsDto
+                    {
+                        Name = x.Name,
+                        PageLink = x.PageLink,
+                        AvailabilitySlots = x.AvailabilitySlots
+                            .Select(y =>
+                                new AvailabilitySlotDto
+                                {
+                                    Id = y.Id,
+                                    Name = y.Name,
+                                    Type = y.Type,
+                                    Size = y.Size,
+                                    IsEnabled = y.IsEnabled,
+                                    Author =
+                                        new UserDto
+                                        {
+                                            UserName = y.Author.Name,
+                                            Image = y.Author.ImagePath,
+                                        },
+                                    Location = new LocationDto
+                                    {
+                                        Name = y.Location.Name,
+                                    },
+                                    Members = y.Members
+                                        .Select(m =>
+                                         new UserDto
+                                         {
+                                             UserName = m.User.Name,
+                                             Image = m.User.ImagePath,
+                                         })
+                                        .ToList()
```

Comment:
You can create simple DTO and use mapper here

Reply:
I simplified dto and mapper in this method